### PR TITLE
feat: add water qualities

### PR DESCRIPTION
**Pull request by Laetta**

### CODE REVIEW

File: src/hydration/bottles.ts [hunk 1]
```diff
@@ -88,14 +95,18 @@ because those items would otherwise become normal bottles without custom model d
 // Fill a bottle
 registerEvent(PlayerInteractEvent, async (event) => {
   if (event.item?.type !== Material.GLASS_BOTTLE) return;
-  if (!event.item.itemMeta.hasCustomModelData()) return; // Default glass bottle
-  const a = event.action;
-  if (a !== Action.RIGHT_CLICK_AIR && a !== Action.RIGHT_CLICK_BLOCK) return;
+  // if (!event.item.itemMeta.hasCustomModelData()) {
```
Comment (bensku) [inline src/hydration/bottles.ts:98]: Why is this commented? Please explain in a comment or remove the dead code.

File: src/hydration/bottles.ts [hunk 2]
```diff
@@ -88,14 +95,18 @@ because those items would otherwise become normal bottles without custom model d
...
+  //   // Default glass bottle
+
+  //   return;
+  // }
+  if (!isRightClick(event.action)) return;
 
   // Didn't compile? Could this be used instead of setCancelled?
-  //event.setUseItemInHand(Result.DENY)
+  //event.setUseItemInHand(Result.DENY);
```
Comment (bensku) [inline src/hydration/bottles.ts:106]: Please re-confirm: does this *still* not compile?

File: src/hydration/water-quality.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  [Biome.LUKEWARM_OCEAN,        'SALTY'],
+  [Biome.DEEP_LUKEWARM_OCEAN,   'SALTY'],
+  [Biome.WARM_OCEAN,            'SALTY'],
+  [Biome.DEEP_WARM_OCEAN,       'SALTY'],
+
+  [Biome.SNOWY_MOUNTAINS,       'CLEAR'],
+  [Biome.MOUNTAINS,             'CLEAR'],
+]);
+
+const CLEAR_WATER_HEIGHT = 110;
```
Comment (bensku) [inline src/hydration/water-quality.ts:42]: This is *minimum* height for clear water in biomes that have it, not a height that automatically makes water clear in all biomes. Please leave a comment indicating so, I was confused for a while.

**Review by bensku — CHANGES_REQUESTED** [review]

File: src/hydration/bottles.ts (hunk 2 above)
Comment (Laetta) [reply src/hydration/bottles.ts:106]: Compiles, but cancelling the event seems like the best approach. Comments will be deleted now

**Pull request merged**